pkg/upload/react-native-ios.go

```diff
@@ -0,0 +1,197 @@
+			buildDirPath = filepath.Join(path, "build")
+
+			if utils.FileExists(buildDirPath) {
+				rootDirPath = filepath.Join(path, "..")
+
+			} else if bundlePath == "" || sourceMapPath == "" {
+				return fmt.Errorf("unable to find bundle files or source maps in within " + path)
+			}
+		}
+
+		// Set a default value for projectRoot if it's not defined
+		if projectRoot == "" {
+			projectRoot = rootDirPath
+		}
+
+		// Check if we're missing any of these parameters
+		if bundlePath == "" || plistPath == "" && (apiKey == "" || versionName == "" || bundleVersion == "") {
+
+			// Check to see if we have the xcworkspacePath
+			if xcworkspacePath == "" {
+				// If not, attempt to locate it in the path/ios/ folder
+				if utils.IsDir(filepath.Join(path, "ios")) {
+					workspacePath, err := utils.FindFolderWithSuffix(filepath.Join(path, "ios"), ".xcworkspace")
+					if err != nil {
+						return err
+					}
+					xcworkspacePath = workspacePath
+				}
+			}
+
+			// Check to see if we have a scheme
+			if scheme == "" {
+				// If not, work it out from the xcworkspace file
+				possibleSchemeName := strings.TrimSuffix(filepath.Base(xcworkspacePath), ".xcworkspace")
+				schemeExists, err := ios.IsSchemeInWorkspace(xcworkspacePath, possibleSchemeName)
+				if err != nil {
+					return err
+				}
+
+				if schemeExists {
+					scheme = possibleSchemeName
+				}
+			}
+
+			// Pull build settings from the xcworkspace file
+			var err error
+			buildSettings, err = ios.GetXcodeBuildSettings(xcworkspacePath, scheme)
+			if err != nil {
+				return err
+			}
+
+			// Check to see if we have the Info.Plist path
+			if plistPath == "" || !utils.FileExists(plistPath) {
+				// If not, we need to build it from build settings values
+				plistPathExpected := filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.InfoPlistPath)
+				if utils.FileExists(plistPathExpected) {
+					plistPath = plistPathExpected
+					log.Info("Found Info.plist at: " + plistPath)
+				} else {
+					log.Info("No Info.plist found at: " + plistPathExpected)
+				}
+			}
+		}
+
+		// Set a sourceMapPath if it's not defined and check that it exists before proceeding
+		if sourceMapPath == "" {
+			sourceMapFileEnvVar := os.Getenv("SOURCEMAP_FILE")
+
+			// Depending on the value of the SOURCEMAP_FILE environment variable, we will either use the build directory or the value of the environment variable to locate the source map file
+			var sourceMapPathToUse string
+			if sourceMapFileEnvVar == "" {
+				sourceMapPathToUse = buildDirPath
+			} else {
+				sourceMapPathToUse = sourceMapFileEnvVar
+			}
+
+			sourceMapPath = filepath.Join(sourceMapPathToUse, "sourcemaps", "main.jsbundle.map")
+			if !utils.FileExists(sourceMapPath) {
+				return errors.New("Could not find a suitable source map file, please specify the path by using --source-map")
+			}
+		}
+
+		// Set a bundlePath if it's not defined and check that it exists before proceeding
+		if bundlePath == "" {
```

Comment:
This might be safer inside the previous if-statement for bundlePath being empty (~L121) as it uses `buildSettings` which is only populated in there.